File: android/app/src/main/java/com/epam/kborodin/bored/data/local/BoredDatabase.kt
package com.epam.kborodin.bored.data.local

import androidx.room.Database
import androidx.room.RoomDatabase
import com.epam.kborodin.bored.data.local.dao.ActionDao
import com.epam.kborodin.bored.data.local.entity.ActionEntity

@Database(
    entities = [
        ActionEntity::class
    ],
    version = 1
)
abstract class BoredDatabase : RoomDatabase() {

    abstract fun actionDao(): ActionDao
}

File: flutter_app/lib/data/local/bored_database.dart
import 'dart:async';

import 'package:floor/floor.dart';
import 'package:flutter_app/data/local/dao/action_dao.dart';
import 'package:flutter_app/data/local/entity/action_entity.dart';
import 'package:injectable/injectable.dart';
import 'package:sqflite/sqflite.dart' as sqflite;

part 'bored_database.g.dart';

@Database(
  version: 1,
  entities: [
    ActionEntity,
  ],
)
abstract class BoredDatabase extends FloorDatabase {
  @preResolve
  ActionDao get actionDao;
}
